### PR TITLE
osu! Difficulty Aim Overhaul: PP Calc Changes

### PR DESCRIPTION
**Pull request by Xexxar**
- [x] Depends on https://github.com/ppy/osu/pull/14845.
- [x] Depends on https://github.com/ppy/osu/pull/14846.
- [x] Depends on https://github.com/ppy/osu/pull/14847

This change serves to facilitate some ppCalc in order to balance the combined effects of all other changes. With this, the aim rework is complete.

AR changes include a revert to the original non object count based AR length bonus, as well as a buff to low AR scaling for aim and a change to the acc curve to make it slightly harsher with low accuracy for aimPP.

### CODE REVIEW

**Review by apollo-dw — COMMENTED** [review]
Fine with the aim accuracy change considering the survey results - sure.

Low AR change graph: https://www.desmos.com/calculator/xh74d5xcan (blue is proposed)

Is there a particular reason you went with removing the object-based buff and opting for reverting to the base buff? Would it be completely unreasonable to balance the `approachRateTotalHitsFactor` as well?

**Comment by Xexxar** [comment]
> Fine with the aim accuracy change considering the survey results - sure.
> 
> Low AR change graph: https://www.desmos.com/calculator/xh74d5xcan (blue is proposed)
> 
> Is there a particular reason you went with removing the object-based buff and opting for reverting to the base buff? Would it be completely unreasonable to balance the `approachRateTotalHitsFactor` as well?

As we discussed off Github, the length component of the AR system was primarily a bandaid solution to handle extreme abuse cases. Since the problem patterns have been nerfed, I decided to rebalance to the original non length based solution. Per our discussion, we both agree that a form of length awareness to this buff is beneficial, so I have added additional lengthBonus based scaling for AR buffs. It's influence is much smaller now, however.

**Comment by Xexxar** [comment]
Removed the global multiplier nerf to pacify the mald.

**Review by MBmasher — DISMISSED** [review]
I still have issues with how AR bonus is done, especially with the length factor, but this is something that can't be fixed until AR is factored into SR, and these changes do make for better balancing, so I'm OK with this being merged

File: osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs [hunk 1]
```diff
@@ -125,10 +123,8 @@ private double computeAimValue()
                 aimValue *= 1.0 + 0.04 * (12.0 - Attributes.ApproachRate);
             }
 
-            aimValue *= approachRateBonus;
-
-            // Scale the aim value with accuracy _slightly_.
-            aimValue *= 0.5 + accuracy / 2.0;
+            // Scale the aim value with accuracy
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs:126]: Pointless comment. Please use comments to explain why, not to reword the operation on the next line. Might as well remove this if there's nothing to say

File: osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs [hunk 2]
```diff
@@ -99,7 +100,9 @@ private void setDistances(double clockRate)
                 // For hitobjects which continue in the direction of the slider, the player will normally follow through the slider,
                 // such that they're not jumping from the lazy position but rather from very close to (or the end of) the slider.
                 // In such cases, a leniency is applied by also considering the jump distance from the tail of the slider, and taking the minimum jump distance.
-                MovementDistance = Math.Min(JumpDistance, tailJumpDistance);
+                // Additional distance is removed based on position of jump relative to slider follow circle radius.
+                // JumpDistance is the distance beyond the s. tailJumpDistance is maximum_slider_radius since the full distance of radial leniency is still possible.
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs:104]: what's s.? slider? please expand

File: osu.Game.Rulesets.Osu/Difficulty/Skills/Aim.cs [hunk 3]
```diff
@@ -90,12 +96,31 @@ private double strainValueOf(DifficultyHitObject current)
...
                     acuteAngleBonus *= 0.5 + 0.5 * (1 - Math.Min(acuteAngleBonus, Math.Pow(calcAcuteAngleBonus(lastLastAngle), 3))); // Penalize acute angles if they're repeated, reducing the penalty as the lastLastAngle gets more obtuse.
-
-                    angleBonus = acuteAngleBonus * acute_angle_multiplier + wideAngleBonus * wide_angle_multiplier; // add the angle buffs together.
                 }
             }
 
-            aimStrain += angleBonus; // Add in angle bonus.
+            if (Math.Max(prevVelocity, currVelocity) != 0)
+            {
+                prevVelocity = (osuLastObj.JumpDistance + osuLastObj.TravelDistance) / osuLastObj.StrainTime; // We want to use the average velocity over the whole object when awarding differences, not the individual jump and slider path velocities.
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Difficulty/Skills/Aim.cs:104]: These lines are too long. put the comments above and potentially split over multiple lines.

**Review by peppy — DISMISSED** [review]

File: osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs (hunk 2 above)
Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs]: Resolved in base PR (https://github.com/ppy/osu/pull/14846).

File: osu.Game.Rulesets.Osu/Difficulty/Skills/Aim.cs (hunk 3 above)
Comment (Xexxar) [reply osu.Game.Rulesets.Osu/Difficulty/Skills/Aim.cs]: Resolved in velocity PR

**Comment by smoogipoo** [comment]
[Spreadsheet](https://docs.google.com/spreadsheets/d/1XONSMA-LTUMuCw3VpZQM2-fo9ZcEsgMp2ekwjZ2QMCA/edit#gid=0) of these changes + all prerequisites versus the changes in https://github.com/ppy/osu/pull/14845. Looks fine to me.

**Review by apollo-dw — APPROVED** [review]
LGTM

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**